testSrc/unit/io/flutter/sdk/FlutterSdkTest.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2017 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.sdk;
+
+import com.intellij.openapi.vfs.VirtualFile;
+import io.flutter.testing.ProjectFixture;
+import io.flutter.testing.TestDir;
+import io.flutter.testing.Testing;
+import org.junit.Rule;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class FlutterSdkTest {
+
+  @Rule
+  public final ProjectFixture fixture = Testing.makeEmptyModule();
+
+  @Rule
+  public final TestDir tmp = new TestDir();
+
+  @Test
+  public void canReadFlutterSdkFromUnopenedProjectInExampleDirectory() throws Exception {
+    final VirtualFile projectDir = tmp.ensureDir("myproject");
+    tmp.ensureDir("myproject/.idea/libraries");
+    tmp.writeFile("myproject/.idea/libraries/Dart_SDK.xml",
```

Comment:
I think it's totally fine to have such hardcoded paths and XMLs in tests. In production code, you normally don't have any need to write such.